integration/healthcheck_test.go

```diff
@@ -67,16 +55,40 @@ func (s *HealthCheckSuite) TestSimpleConfiguration(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(resp.StatusCode, checker.Equals, 200)
 
-	healthReq, err := http.NewRequest("POST", "http://"+whoami1Host+"/health", bytes.NewBuffer([]byte("500")))
+	// Fix all whoami health to 500
+	whoamiHosts := []string{whoami1Host, whoami2Host}
+	for _, whoami := range whoamiHosts {
+		healthReq, err := http.NewRequest("POST", "http://"+whoami+"/health", bytes.NewBuffer([]byte("500")))
+		c.Assert(err, checker.IsNil)
+		_, err = client.Do(healthReq)
+		c.Assert(err, checker.IsNil)
+	}
+
+	try.Sleep(2 * time.Second)
+
+	// Verify frontend health : 500
+	err = try.Request(req, 3*time.Second, try.StatusCodeIs(500))
 	c.Assert(err, checker.IsNil)
-	_, err = client.Do(healthReq)
+
+	// Change one whoami health to 200
+	healthReq1, err := http.NewRequest("POST", "http://"+whoami1Host+"/health", bytes.NewBuffer([]byte("200")))
+	c.Assert(err, checker.IsNil)
+	_, err = client.Do(healthReq1)
 	c.Assert(err, checker.IsNil)
 
-	time.Sleep(time.Second * 3)
+	// Verify frontend health : before
+	err = try.Request(req, 3*time.Second, try.StatusCodeIs(500))
+	c.Assert(err, checker.IsNil)
 
-	resp, err = client.Do(req)
+	try.Sleep(2 * time.Second)
+
+	// Verify frontend health : after
```

Comment:
I think the block starting from here to line 91 (incl.) isn't needed as we just want the single, static request starting at line 93.